CRM/Contribute/Form/Contribution/Confirm.php

```diff
@@ -1537,11 +1537,21 @@ protected function postProcessMembership(
           $pending = $membershipContribution->contribution_status_id == CRM_Core_PseudoConstant::getKey('CRM_Contribute_BAO_Contribution', 'contribution_status_id', 'Pending');
         }
         else {
-          $pending = $this->getIsPending();
+          // The concept of contributeMode is deprecated.
+          // the is_monetary concept probably should be too as it can be calculated from
+          // the existence of 'amount' & seems fragile.
+          if (((isset($this->_contributeMode)) || !empty($this->_params['is_pay_later'])
+            ) &&
+            (($this->_values['is_monetary'] && $this->_amount > 0.0))
+          ) {
+            $pending = TRUE;
+          }
+          $pending = FALSE;
```

Comment:
This function is only used here. We want to move it to BAO/membership but first step is remove function and put inline so you can see what it is doing.
This is the only bit that is not replacing `CRM_Utils_Array::value`